src/netCDF4/__init__.pyi

```diff
@@ -474,15 +615,14 @@ class Variable(Generic[T_Datatype]):
     def set_ncstring_attrs(self, ncstring_attrs: bool) -> None: ...
     def set_collective(self, value: bool) -> None: ...
     def get_dims(self) -> tuple[Dimension, ...]: ...
-
     def __delattr__(self, name: str) -> None: ...
     def __setattr__(self, name: str, value: Any) -> None: ...
     def __getattr__(self, name: str) -> Any: ...
-    def __getitem__(self, elem: GetSetItemKey) -> np.ndarray: ...
+    def __getitem__(self, elem: GetSetItemKey) -> Any: ...
```

Comment:
Why did you change the return type here?
I feel like having this as np.ndarray is quite advantageous for end user code.

Reply:
Primarily because of the case of `str`. If you index a singular string value, the return type is `str`. E.g.

```python
# given "strvar" of type str and shape (20, 3)
strvar[0, 0] = "abc"
value = strvar[0, 0]
assert type(value) == str  # OK
```

So the return type either has to be a union of the two (I don't think it's solvable with overloads) or `Any`.

Reply:
Furthermore if `var.auto_mask` is `False`, this may return values and not arrays. E.g.

```python
# given variable intvar of type int64 and shape (20, 3)
intvar.set_auto_mask(False)
intvar[...] = np.arange(60).reshape(20, 3)
assert type(intvar[0, 0]) == np.int64  # OK
```

Reply:
The dimensions in these examples don't really matter, they're just what I used when I was playing around with the idea.